### ISSUE TITLE
ansible-test duplicate-code with import future lines

### ISSUE THREAD

**Issue by rrey** [ansible/ansible#65816]
##### SUMMARY

`ansible-test` is reported duplicated lines in my collections while running pylon test.
The duplicated lines are the `from __future__ import ...` that are mandatory according to other pylint tests.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ansible-test

##### ANSIBLE VERSION

```paste below
 ansible 2.9.2
   config file = None
   configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
   ansible python module location = /usr/lib/python2.7/site-packages/ansible
   executable location = /usr/bin/ansible
   python version = 2.7.5 (default, Aug  7 2019, 00:51:29) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### OS / ENVIRONMENT

`ansible-test` command is executed from gitlab-ci using a centos:7 image.

##### STEPS TO REPRODUCE

```yaml
ansible-test sanity --docker
```

##### EXPECTED RESULTS

No error just like a local run with `--docker`

##### ACTUAL RESULTS

```paste below
Running sanity test 'pslint'
 Running sanity test 'pylint' with Python 3.6
 ERROR: Found 1 pylint issue(s) which need to be resolved:
 ERROR: plugins/modules/harbor_user.py:1:0: duplicate-code: Similar lines in 2 files ==harbor_project:6 ==harbor_user:6 from __future__ import (absolute_import, division, print_function)  ANSIBLE_METADATA = {     'status': ['preview'],     'supported_by': 'community',     'metadata_version': '1.1' }  DOCUMENTATION = ''' ---
 See documentation for help: https://docs.ansible.com/ansible/2.9/dev_guide/testing/sanity/pylint.html
 Running sanity test 'replace-urlopen' with Python 3.6
 Running sanity test 'rstcheck' with Python 3.6
 Running sanity test 'shebang' with Python 3.6
 Running sanity test 'shellcheck'
```

**Comment by ansibot** [comment]
Files identified in the description:
* [`bin/ansible-test`](https://github.com/ansible/ansible/blob/devel/bin/ansible-test)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gundalow** [comment]
@rrey Thank you for this but report, could you please paste in the first 10 lines of:
* `harbor_user`
* `harbor_project`

As the error message is confusing as it's talking about 
 ERROR: plugins/modules/**harbor_user.py:1:0**: duplicate-code: Similar lines in 2 files ==**harbor_project:6** ==**harbor_user:6** from __future__ import (absolute_import, division, print_function)  ANSIBLE_METADATA = {     'status': ['preview'],     'supported_by': 'community',     'metadata_version': '1.1' }  DOCUMENTATION = ''' ---

**Comment by samdoran** [comment]
needs_info

**Comment by ansibot** [comment]
@rrey This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by rrey** [comment]
Sorry for the delay. Here are the 10 first lines of each files:

```
$ head -10 harbor_user.py harbor_project.py 
==> harbor_user.py <==
#!/usr/bin/python
# -*- coding: utf-8 -*-
#
# Copyright: (c) 2019, SFR
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)

from __future__ import (absolute_import, division, print_function)

ANSIBLE_METADATA = {
    'status': ['preview'],

==> harbor_project.py <==
#!/usr/bin/python
# -*- coding: utf-8 -*-
#
# Copyright: (c) 2019, SFR
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)

from __future__ import (absolute_import, division, print_function)

ANSIBLE_METADATA = {
    'status': ['preview'],
```

**Comment by mattclay** [comment]
@rrey I'm not able to reproduce this issue locally. Could you provide a sample collection that triggers the issue, along with the full command you're running?

**Comment by rrey** [comment]
@mattclay we are in discussion to release the code. It dont’t happen before a week or two but I’ll port it to gitlab.com to reproduce the issue on a public repo.

**Comment by pabelanger** [comment]
I can't figure it out, but also seeing this both on devel and latest 2.9.

http://paste.openstack.org/show/794220/ is pip freeze on devel, this is being run under python3.6 on centos-8

**Comment by pabelanger** [comment]
For now, we have used
```
# pylint: skip-file

as there is no other way to disable duplicate-code checks: https://github.com/PyCQA/pylint/issues/214

**Comment by mattclay** [comment]
@rrey @pabelanger Have you had any success creating a reproducer for this issue?

**Comment by pabelanger** [comment]
Yes, we can reproduce it will zuul. We can hold a node and allow someone to inspect it to help this issue. We'll just need the public SSH key of the user.

**Comment by sivel** [comment]
@pabelanger I've pinged you with a key on slack.  I can try to take a look.

Although I'm beginning to question whether we should just add the ignore and move on. It's not necessarily something we are concerned with, since many modules will have some amount of duplication anyway.

**Comment by mattclay** [comment]
One of the goals of ansible-test is consistent test results across different environments. That's why I'm interested in figuring out why we're seeing different test results in this case.

**Comment by sivel** [comment]
Alright, I've tracked this down, to the fact that we set `--jobs 0` and the zuul VMs have only a single core/vcpu.

With `--jobs 0` it selects the number of cores in a machine, and when set at `1` pylint does all processing in the launched process. When `--jobs > 1` then the launched process spawns up to `--jobs` number of workers, and the launched process is just the controller.

So normally for most devs, we are operating on multi core hosts, so a single checker never persists past a single file/module specified on the CLI.

I'm thinking we should instead calculate the job count using `N+1`, instead of relying on pylint. Which means that regardless of system cores, the behavior should be the same.
